scs2-simulation/src/main/java/us/ihmc/scs2/simulation/physicsEngine/contactPointBased/RobotOneDoFJointSoftLimitCalculator.java

```diff
@@ -39,6 +45,19 @@ public void compute()
       }
    }
 
+   public void compute(DMatrix tauToAppendTo)
+   {
+      compute();
+
+      for (JointCalculator calculator : jointCalculators)
+      {
+         int jointIndex = jointMatrixIndexProvider.getJointDoFIndices(calculator.joint)[0];
+         double currentValue = tauToAppendTo.get(jointIndex, 0);
+         // Appending this tau to the current value
+         tauToAppendTo.set(jointIndex, 0, currentValue + calculator.jointLimitEffort.getDoubleValue());
```

Comment:
Kinda dookie that we have to iterate again. I wonder if we should consider passing the `YoMatrix` in the constructor, and then each individual `JointCalculator` can grab the `YoVariable` it'll modify, so we don't have to iterate anymore. Downside is that it make this less readable I suppose

Reply:
Right, after passing over this again, I don't think we need to iterate twice either. I've moved the `calculator`-wise `compute()` method inside this method. So now, the only reason I'm keeping `Robot...Calculator.compute()` around is for compatability with the impulse-based physics engine, which I haven't touched yet. If we agree my overall approach is good for the contact-based physics engine, I'll do the same for the impulse-based one, and then we can completely remove the `Robot...Calculator.compute()` above in favour of `Robot...Calculator.compute(DMatrix)`

Reply:
Now that I've had a pass at the impulse-based stuff, the redundant `compute()` method is removed, we only use `compute(DMatrix)` and iterate once. Same for the other `RobotOneDoFJointDampingCalculator`